### PR TITLE
Add mouse WheelLeft/WheelRight buttons

### PR DESCRIPTION
**Pull request by rrPKrr**
Add ability to bind commands to mouse wheel left/mouse wheel right buttons.

### CODE REVIEW

File: src/m_menu.c [hunk 1]
```diff
@@ -5830,7 +5830,7 @@ boolean M_Responder (event_t* ev)
 		// Don't bind movement and turning to mouse wheel. It needs to
 		// be impossible to input a one-frame of movement automatically
 		// in speedrunning.
-		if ((ch == MOUSE_BUTTON_WHEELUP || ch == MOUSE_BUTTON_WHEELDOWN) &&
```
Comment (kraflab) [inline src/m_menu.c:5833]: Maybe it's time for `M_IsMouseWheel(int mouseb)` in `m_input`? :^)

File: src/i_video.c [hunk 2]
```diff
@@ -497,14 +497,22 @@ static void MapMouseWheelToButtons(SDL_MouseWheelEvent *wheel)
     static event_t down;
     int button;
 
-    if (wheel->y <= 0)
+    if (wheel->y < 0)
```
Comment (fabiangreffrath) [inline src/i_video.c:500]: We'll lose the hypothetical (maybe not so?) case that `wheel` is all zero here. Please do either initialize `button` to something reasonable, or keep the `<=` checks (preferred) or add an `else return` clause.

File: src/m_menu.c [hunk 3]
```diff
@@ -5830,7 +5830,7 @@ boolean M_Responder (event_t* ev)
...
+		if ((ch == MOUSE_BUTTON_WHEELUP || ch == MOUSE_BUTTON_WHEELDOWN || ch == MOUSE_BUTTON_WHEELLEFT || ch == MOUSE_BUTTON_WHEELRIGHT) &&
 		    s_input >= input_forward && s_input <= input_straferight)
```
Comment (fabiangreffrath) [inline src/m_menu.c:5834]: These lines are getting to long. Please add a line break after the added `||`.

File: src/m_misc.c [hunk 4]
```diff
@@ -2738,7 +2738,7 @@ boolean M_ParseOption(const char *p, boolean wad)
                   // Don't bind movement and turning to mouse wheel. It needs to
                   // be impossible to input a one-frame of movement
                   // automatically in speedrunning.
-                  if ((value == MOUSE_BUTTON_WHEELUP || value == MOUSE_BUTTON_WHEELDOWN) &&
+                  if ((value == MOUSE_BUTTON_WHEELUP || value == MOUSE_BUTTON_WHEELDOWN || value == MOUSE_BUTTON_WHEELLEFT || value == MOUSE_BUTTON_WHEELRIGHT) &&
                       dp->ident >= input_forward && dp->ident <= input_straferight)
```
Comment (fabiangreffrath) [inline src/m_misc.c:2742]: And also here

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: src/m_menu.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/m_menu.c:5833]: > Maybe it's time for `M_IsMouseWheel(int mouseb)` in `m_input`? :^)

Good idea, but can probing wait till after this PR.

File: src/i_video.c (hunk 2 above)
Comment (fabiangreffrath) [reply src/i_video.c:500]: No, wait. Please add an `else return` clause.

File: src/m_menu.c [hunk 5]
```diff
@@ -5830,8 +5830,9 @@ boolean M_Responder (event_t* ev)
 		// Don't bind movement and turning to mouse wheel. It needs to
 		// be impossible to input a one-frame of movement automatically
 		// in speedrunning.
-		if ((ch == MOUSE_BUTTON_WHEELUP || ch == MOUSE_BUTTON_WHEELDOWN) &&
-		    s_input >= input_forward && s_input <= input_straferight)
+		if ((ch == MOUSE_BUTTON_WHEELUP || ch == MOUSE_BUTTON_WHEELDOWN || 
+         ch == MOUSE_BUTTON_WHEELLEFT || ch == MOUSE_BUTTON_WHEELRIGHT) &&
```
Comment (fabiangreffrath) [inline src/m_menu.c:5834]: Indentation somehow got borked here and in the next line.

Comment (rrPKrr) [reply src/m_menu.c:5834]: Huh, that's strange... Formatting seems to be correct on my end, but the commit looks borked indeed.

**Comment by fabiangreffrath** [comment]
Thank you very much!

**Pull request merged**